GTViewController/Classes/Nav Classes/GTFollowupViewController.m

```diff
@@ -296,4 +303,44 @@ - (void)alertView:(UIAlertView *)alertView clickedButtonAtIndex:(NSInteger)butto
     [alertView dismissWithClickedButtonIndex:buttonIndex animated:YES];
 }
 
+
+#pragma mark - UISnuffleButtonTapDelegate methods
+
+- (BOOL)validateFields {
+    NSArray *inputValidationErrors = [self inputValidationErrors];
+
+    if ([inputValidationErrors count]) {
+        [[[UIAlertView alloc] initWithTitle:[[GTFileLoader sharedInstance] localizedString:@"GTFollowupViewController_validation_title"]
+                                    message:[inputValidationErrors componentsJoinedByString:@"\n"]
+                                   delegate:self
+                          cancelButtonTitle:@"Ok"
+                          otherButtonTitles:nil] show];
+        return NO;
+    }
+
+    return YES;
+}
+
+- (CGRect)pageFrame {
+    return self.followupModalView.frame;
+}
+
+- (UIView *)viewOfPageViewController {
+    return self.followupModalView;
+}
+
+- (NSArray *)inputValidationErrors {
+    __block NSMutableArray *validationErrors = @[].mutableCopy;
+    
+    [self.followupModalView.inputFieldViews enumerateObjectsUsingBlock:^(id  _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop) {
```

Comment:
This doesn't need to change but just as a reminder, you can add the variable casting in this line:
`[self.followupModalView.inputFieldViews enumerateObjectsUsingBlock:^(GTInputFieldView *inputFieldView, NSUInteger idx, BOOL * _Nonnull stop) {`